controllers/pokemon.go

```diff
@@ -0,0 +1,115 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"challenge/api/files"
+	"challenge/api/structs"
+
+	"github.com/gorilla/mux"
+)
+
+var pokemons []structs.Pokemon
+var concurrencyPokemons []structs.Pokemon
+var wg sync.WaitGroup
+var mut sync.Mutex
+
+func initPokemons() {
+	fmt.Println("Set all pokemons")
+	pokemons = nil
+	pokemons = append(pokemons, files.GetPokemonInfoCSV()...)
+}
+
+func HomePage(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("<h1>Welcome to my Pokemon API with golang</h1>"))
+}
+
+func GetPokemons(w http.ResponseWriter, r *http.Request) {
+	getPokemonsExternal()
+	initPokemons()
+	fmt.Println("Get pokemons")
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(pokemons)
+}
+
+func GetPokemon(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("Get pokemon")
+	w.Header().Set("Content-Type", "application/json")
+	// grab id from request
+	params := mux.Vars(r)
+	name := params["name"]
+	pokemon, err := findPokemonByName(name)
+
+	if err != nil {
+		http.Error(w, "Pokemon not found", http.StatusNotFound)
+		return
+	}
+
+	json.NewEncoder(w).Encode(pokemon)
+}
+
+func GetNextPokemons(w http.ResponseWriter, r *http.Request) {
+	getNextPokemonsExternal()
+	initPokemons()
+	fmt.Println("Get next pokemons")
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(pokemons)
+}
+
+func GetPreviousPokemons(w http.ResponseWriter, r *http.Request) {
+	getPreviousPokemonsExternal()
+	initPokemons()
+	fmt.Println("Get previous pokemons")
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(pokemons)
+}
+
+func ConcurrencyGetPokemons(w http.ResponseWriter, r *http.Request) {
+	initPokemons()
+	fmt.Println("Get Concurrency pokemons")
+	w.Header().Set("Content-Type", "application/json")
+
+	var query structs.Query
+	rawQuery := r.URL.Query()
+	query.SetValues(rawQuery)
+
+	if !query.ValidateType() {
```

Comment:
If the items and items_per_worker are not set, you will get `null`
You should display a valid error, like in the `type` validaiton